Automattic/wp-calypso
# Zoninator Extension: update `zones` data handlers to dispatchRequestEx

Migrates `zones` data layer handlers to `dispatchRequestEx`.

Code that converts server responses (individual zone and list of zones) to Redux structures is moved to `*FromApi` functions in `utils.js`. The zones list transform is rewritten from `reduce` to a very simple `map` + `keyBy`.

Most tests are removed because they duplicate the implementation so much so they are completely useless. The only interesting ones remaining are the `*FromApi` tests.

#### Testing instructions

Install the Zoninator plugin on your Jetpack or Atomic site and then try to do CRUD operations on the zones list in the Calypso UI. Are REST requests being issued and UI updated?

See the testing instructions in #27415 for more details about the Zoninator setup. I might be a good idea to test both PRs at the same time :wink:
